AllJoyn/Platform/BridgeHost/App.xaml.cs

```diff
@@ -0,0 +1,359 @@
+    /// </summary>
+    sealed partial class App : Application
+    {
+        /// <summary>
+        /// Initializes the singleton application object.  This is the first line of authored code
+        /// executed, and as such is the logical equivalent of main() or WinMain().
+        /// </summary>
+        public App()
+        {
+            this.InitializeComponent();
+            this.Construct();
+        }
+
+        private AppServiceConnection appServiceConnection;
+        private BackgroundTaskDeferral appServiceDeferral;
+        private const string OnboardedDevicesContainerName = "OnboardedDevices";
+        public static bool isStartUpTaskRunning = false;
+        /// <summary>
+        /// Initializes the app service on the host process 
+        /// </summary>
+        protected async override void OnBackgroundActivated(BackgroundActivatedEventArgs args)
+        {
+            Debug.WriteLine("OnBackgroundActivated ");
+            base.OnBackgroundActivated(args);
+
+            IBackgroundTaskInstance taskInstance = args.TaskInstance;
+            string taskName = taskInstance.Task.Name;
+            Debug.WriteLine("TaskName = " + taskName);
+            var triggerDetails = args.TaskInstance.TriggerDetails as AppServiceTriggerDetails;
+            bool isStartupTask = triggerDetails.CallerPackageFamilyName.Equals(Windows.ApplicationModel.Package.Current.Id.FamilyName);
+
+            if (isStartupTask ||
+                IsSupportedAppServiceClient(triggerDetails.CallerPackageFamilyName))
+            {
+                appServiceDeferral = taskInstance.GetDeferral();
+                taskInstance.Canceled += TaskInstance_Canceled;
+
+                var appServiceDetails = triggerDetails as AppServiceTriggerDetails;
+                Debug.WriteLine("New Connection from " + appServiceDetails.CallerPackageFamilyName);
+                appServiceConnection = appServiceDetails.AppServiceConnection;
+                appServiceConnection.RequestReceived += AppServiceConnection_RequestReceived;
+                appServiceConnection.ServiceClosed += AppServiceConnection_ServiceClosed;
+            }
+
+            if (isStartupTask)
+            {
+                isStartUpTaskRunning = true;
+            }
+            else
+            {
+                StartOpenT2TAppServiceAnchorProcess(); // Background process is not yet launched. Launch it so that the reference is kept alive.
+            }
+
+            // Onboard the devices that were onboarded previously.
+            await OnboardRegisteredDevices();
+        }
+
+        private bool IsSupportedAppServiceClient(string callerPackageFamilyName)
+        {
+            return callerPackageFamilyName.Equals("org.opent2t.console_h35559jr9hy9m", StringComparison.OrdinalIgnoreCase) || // OpenT2T Console app
+                callerPackageFamilyName.Equals("OpenT2TAllJoynBridgeUWPTestClient_a6049pvczh34w", StringComparison.OrdinalIgnoreCase); // Test app
+        }
```

Comment:
Just two and I would leave it like this for now. 

---
In reply to: [75184473](https://github.com/tjaffri/msiot-samples/pull/9#discussion_r75184473) [](ancestors = 75184473)